testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/messaging/jms/external/prefix/ExternalJMSDestinationDefinitionMessagingDeploymentTestCase.java

```diff
@@ -0,0 +1,304 @@
+public class ExternalJMSDestinationDefinitionMessagingDeploymentTestCase {
+
+    public static final String QUEUE_LOOKUP = "java:/jms/AutomaticQueueCreationOnExternalMessagingDeploymentTestCase/myQueue";
+    public static final String TOPIC_LOOKUP = "java:/jms/AutomaticQueueCreationOnExternalMessagingDeploymentTestCase/myTopic";
+    public static final String REMOTE_PCF = "remote-artemis";
+
+    public static final String QUEUE_NAME = "myExternalQueue";
+    public static final String TOPIC_NAME = "myExternalTopic";
+    private static Set<String> initialQueues = null;
+
+    static class SetupTask implements ServerSetupTask {
+
+        private static final Logger logger = Logger.getLogger(ExternalJMSDestinationDefinitionMessagingDeploymentTestCase.SetupTask.class);
+        private final Map<String, AutoCloseable> snapshots = new HashMap<>();
+
+        @Override
+        public final void setup(ManagementClient managementClient, String containerId) throws Exception {
+            snapshots.put(containerId, ServerSnapshot.takeSnapshot(managementClient));
+            Set<String> runtimeQueues = listRuntimeQueues(managementClient);
+            ServerReload.executeReloadAndWaitForCompletion(managementClient, true);
+            JMSOperations ops = JMSOperationsProvider.getInstance(managementClient.getControllerClient());
+            boolean needRemoteConnector = ops.isRemoteBroker();
+            if (needRemoteConnector) {
+                ops.addExternalRemoteConnector("remote-broker-connector", "messaging-activemq");
+            } else {
+                ops.addExternalHttpConnector("http-test-connector", "http", "http-acceptor");
+            }
+            //To avoid security limitations as default role guest doesn't have the 'manage' rights
+            // /subsystem=messaging-activemq/server=default/security-setting=#/role=guest:write-attribute(name=manage, value=true)
+            ops.disableSecurity();
+            ModelNode op = Operations.createRemoveOperation(getInitialPooledConnectionFactoryAddress(ops.getServerAddress()));
+            managementClient.getControllerClient().execute(op);
+            op = Operations.createAddOperation(getPooledConnectionFactoryAddress());
+            op.get("transaction").set("xa");
+            op.get("entries").add("java:/JmsXA java:jboss/DefaultJMSConnectionFactory");
+            if (needRemoteConnector) {
+                op.get("connectors").add("remote-broker-connector");
+            } else {
+                op.get("connectors").add("http-test-connector");
+            }
+            op.get("enable-amq1-prefix").set(false);
+            execute(managementClient, op, true);
+            op = Operations.createRemoveOperation(getClientTopicAddress());
+            managementClient.getControllerClient().execute(op);
+            op = Operations.createRemoveOperation(getClientQueueAddress());
+            managementClient.getControllerClient().execute(op);
+            ops.close();
```

Comment:
Neither `ActiveMQProviderJMSOperations.java`  nor `RemoteActiveMQProviderJMSOperations.java` implements this `close()` method 
```
    @Override
    public void close() {
        // no-op
        // DO NOT close the management client. Whoever passed it into the constructor should close it
    }
```

but I think it won't do any harm.

Reply:
yes i agree